src/app/containers/StakePage/components/HistoryEventsTable.tsx

```diff
@@ -222,35 +153,42 @@ interface HisoryAsset {
   index: number;
 }
 
-const HisoryTableAsset: React.FC<HisoryAsset> = ({ item, index }) => {
+const HisoryTableAsset: React.FC<HisoryAsset> = ({ item }) => {
   const { t } = useTranslation();
   const SOV = AssetsDictionary.get(Asset.SOV);
   const dollars = useCachedAssetPrice(Asset.SOV, Asset.USDT);
   const dollarValue = useMemo(() => {
-    if (item.returnVal.amount === undefined) return '';
-    return bignumber(item.returnVal.amount)
+    if (item.amount === undefined || null) return '';
+    return bignumber(item.amount)
       .mul(dollars.value)
       .div(10 ** SOV.decimals)
       .toFixed(0);
-  }, [dollars.value, item.returnVal.amount, SOV.decimals]);
+  }, [dollars.value, item.amount, SOV.decimals]);
   return (
     <tr>
       <td>
         {moment
-          .tz(new Date(item.eventDate), 'GMT')
+          .tz(new Date(item.time * 1e3), 'GMT')
           .format('DD/MM/YYYY - h:mm:ss a z')}
       </td>
+      <td>{item.action}</td>
       <td className="tw-text-left tw-font-normal">
-        {numberFromWei(item.returnValues.amount)} SOV
-        <br />≈{' '}
-        <LoadableValue
-          value={numberToUSD(Number(weiToFixed(dollarValue, 4)), 4)}
-          loading={dollars.loading}
-        />
+        {item.action !== t(translations.stake.actions.delegate) ? (
+          <>
+            {numberFromWei(item.amount)} SOV
+            <br />≈{' '}
+            <LoadableValue
+              value={numberToUSD(Number(weiToFixed(dollarValue, 4)), 4)}
```

Comment:
Use our `weiTo4` method instead of `weiToFixed(..., 4)`.

Reply:
fixed in #1438